### ISSUE TITLE
Returns addressbook as vcard when addressbook is empty

### ISSUE THREAD

**Issue by chmac** [natelindev/tsdav#75]
Steps to reproduce:

- Create an empty address book on the CardDAV server
- Use `fetchVCards()` against it
- Note that you receive 1 record which has the address of the address book and not a VCard

Having looked into it, a little (by uncommenting the code that logs xml requests and responses), it looks like the result of the `addressbook-query` is an empty set, but somehow in this code that is converted into a URL for the address book itself:

https://github.com/natelindev/tsdav/blob/0546cc0dc70870022df85228461fb49653d793a0/src/addressBook.ts#L136-L150

Then in the next request, the `addressbook-multiget`, the request looks like:

```
<card:addressbook-multiget xmlns:d="DAV:" xmlns:card="urn:ietf:params:xml:ns:carddav">
  <d:prop>
    <d:getetag />
    <card:address-data />
  </d:prop>
  <d:href>/user/contacts/abook/</d:href>
</card:addressbook-multiget>
```

Where the `d:href` is the address book itself.

The `addressbook-query` always looks like:

```xml
<?xml version="1.0" encoding="utf-8"?>
<card:addressbook-query xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:d="DAV:">
  <d:prop>
    <d:getetag />
  </d:prop>
  <card:filter>
    <card:prop-filter name="FN" />
  </card:filter>
</card:addressbook-query>
```

Then the response, when the collection is empty, from xandikos, is:

```xml
<ns0:multistatus xmlns:ns0="DAV:" />
```

I suspect that what's happening is online 146 above, the empty response is being turned into an empty string. Then lines 148 and 149 turn that empty string into the URL of the address book. But the filter on line 150 is comparing a full URL against a path. At least that's what my `debug()` output shows when I log those values.

I'd guess that the filter will always fail, because line 149 always converts URLs into paths.

Debugging further, it looks like inside `davRequest` if the response is empty, as per the XML above, then a single element gets returned like so:

```js
[ { status: 207, statusText: 'Multi-Status', ok: true } ]
```

While I guess that an empty array would be more appropriate, as the response is empty. Okay, I think I figured it out, I'll submit a PR right after this.

**Comment by natelindev** [comment]
Thanks for the contribution, this is a rather complex bug, I'll publish a new version today addressing this.

**Comment by natelindev** [comment]
fixed in v2.0.0-rc.3

**Comment by natelindev** [comment]
additional fix will be applied in release v2.0.0
